pkg/reconciler/common/deployments.go

```diff
@@ -50,6 +50,7 @@ func CheckDeployments(ctx context.Context, manifest *mf.Manifest, instance v1alp
 		}
 	}
 	status.MarkDeploymentsAvailable()
+	status.SetVersion(TargetVersion(instance))
```

Comment:
I would prefer we not do this here, for the same reason we shouldn't do it after the install: some temporary error in `DeleteObsoleteResources` might prevent a subsequent reconcile to delete the obsolete resources after the temporary error condition is resolved. Instead, I think we should set `status.Version` in the reconciler itself, as the very last stage.